test/src/graph/nodes/CompactPointsNodeTest.cpp

```diff
@@ -1,23 +1,229 @@
+		ASSERT_THAT(compactNodes.at(0), testing::NotNull());
+		ASSERT_RGL_SUCCESS(rgl_graph_node_add_child(usePointsNode, compactNodes.at(0)));
+		ASSERT_RGL_SUCCESS(rgl_graph_run(usePointsNode));
+	}
 };
 
+INSTANTIATE_TEST_SUITE_P(CompactPointsNodeTests, CompactPointsNodeTest, testing::Values(1, 100, maxGPUCoresTestCount),
+                         [](const auto& info) { return "pointsCount_" + std::to_string(info.param); });
+
 TEST_F(CompactPointsNodeTest, invalid_argument_node)
 {
 	EXPECT_RGL_INVALID_ARGUMENT(rgl_node_points_compact(nullptr), "node != nullptr");
 }
 
-TEST_F(CompactPointsNodeTest, valid_arguments)
+TEST_F(CompactPointsNodeTest, valid_argument_node_is_nullptr)
+{
+	EXPECT_RGL_SUCCESS(rgl_node_points_compact(&compactNodes.at(0)));
+	EXPECT_THAT(compactNodes.at(0), testing::NotNull());
+}
+
+TEST_F(CompactPointsNodeTest, valid_argument_node_is_not_nullptr)
+{
+	ASSERT_RGL_SUCCESS(rgl_node_points_compact(&compactNodes.at(0)));
+	ASSERT_THAT(compactNodes.at(0), testing::NotNull());
+
+	// If (*compactNodes.at(0)) != nullptr
+	EXPECT_RGL_SUCCESS(rgl_node_points_compact(&compactNodes.at(0)));
+}
+
+TEST_F(CompactPointsNodeTest, invalid_pipeline_when_no_input_node)
 {
-	EXPECT_RGL_SUCCESS(rgl_node_points_compact(&compactNode));
-	ASSERT_THAT(compactNode, testing::NotNull());
+	ASSERT_RGL_SUCCESS(rgl_node_points_compact(&compactNodes.at(0)));
+	EXPECT_RGL_INVALID_PIPELINE(rgl_graph_run(compactNodes.at(0)), "looked for IPointsNode");
+}
+
+TEST_F(CompactPointsNodeTest, invalid_pipeline_when_incorrect_input_node)
+{
+	std::vector<rgl_mat3x4f> rayTf;
+	rayTf.emplace_back(Mat3x4f::identity().toRGL());
+	ASSERT_RGL_SUCCESS(rgl_node_rays_from_mat3x4f(&useRaysNode, rayTf.data(), rayTf.size()));
+
+	ASSERT_RGL_SUCCESS(rgl_node_points_compact(&compactNodes.at(0)));
+	ASSERT_RGL_SUCCESS(rgl_graph_node_add_child(useRaysNode, compactNodes.at(0)));
+	EXPECT_RGL_INVALID_PIPELINE(rgl_graph_run(compactNodes.at(0)), "looked for IPointsNode");
+}
+
+TEST_P(CompactPointsNodeTest, points_all_non_hit)
+{
+	int pointsCount = GetParam();
+
+	setUpPointCloud(pointsCount, genCoord, genAllNonHit);
+	runGraphWithAssertions();
+
+	outPointCloud = std::make_unique<TestPointCloud>(TestPointCloud::createFromNode(compactNodes.at(0), pointFields));
+	EXPECT_EQ(outPointCloud->getPointCount(), 0);
+
+	// Check if the contents of outData have changed (they should not have)
+	std::vector<::Field<XYZ_F32>::type> outData{
+	    { 2.0f,  3.0f,  5.0f},
+        { 7.0f, 11.0f, 13.0f},
+        {17.0f, 19.0f, 23.0f}
+    };
+	auto outDataCopy = outData;
+
+	ASSERT_RGL_SUCCESS(rgl_graph_get_result_data(compactNodes.at(0), RGL_FIELD_XYZ_F32, outData.data()));
+	ASSERT_EQ(outData.size(), outDataCopy.size());
+	for (int i = 0; i < outData.size(); ++i) {
+		EXPECT_EQ(outData.at(i)[0], outDataCopy.at(i)[0]);
+		EXPECT_EQ(outData.at(i)[1], outDataCopy.at(i)[1]);
+		EXPECT_EQ(outData.at(i)[2], outDataCopy.at(i)[2]);
+	}
```

Comment:
1. Add a check that `rgl_graph_get_result_size` is 0. 
2. Add a comment that you are verifying that no overwrite is happening, it's not obvious at first sight.